### ISSUE TITLE
Reduce download size when updating NVDA

### ISSUE THREAD

**Issue by nvaccessAuto** [nvaccess/nvda#3328]
**Edit**
The size of NVDA updates could be reduced by using a patch / diff. This will make updates faster on slow internet connections.

**Originally Reported by Animesh.mr on 2013-07-03 14:37**
```
hi all i think n v  d a should download patches not the full setup this will make update feature much more nice it will take less time to update mostly with slow internet connections
i am sure that we can do this 
**Blocking #5148**
```

**Comment by feerrenrut** [comment]
This will add complexity to the update process, and put pressure on the update server. Given the relatively small size of NVDA this issue should be considered a low priority.

**Comment by Adriani90** [comment]
cc: @gerald-hartig, @seanbudd what is the position of NV Access on this meanwhile? I think the size of NVDA installer / updates is quite low anyway, but given the current roadmap there might be updates which could lead to retriaging this issue.

**Comment by gerald-hartig** [comment]
@Adriani90 my take is that this is a nice to have thing, but that currently we have higher priorities. Slowest 10 percentile of internet speeds is currently around 2 mbps. That's a 2 minute download for NVDA, and 90% of the population would be able to do it faster than that.

That being said, if someone wants to make a proof of concept to show how simple it would be, we don't want to discourage that. The lower the cost to implement, the more likely it will make it in. As you mentioned the roadmap document, it's entirely plausible that our download size will increase at some point in the future. And if we move away from our installer / packager, this issue should be taken into consideration when looking at alternatives.

**Comment by gerald-hartig** [comment]
Closing issue. In the 10 years since this was opened it's not become clear that there's a compelling use case for the additional complexity. This might change in the future if the size of the installer increases dramatically, in which case we can revisit this.
